tpm2/Makefile

```diff
@@ -351,100 +385,134 @@ doc: $(HEADER_FILES) \
 	doxygen Doxyfile
 
 .PHONY: test
-test: pretest kmyth-test
+test: pretest clean-backups kmyth-test
 	./bin/kmyth-test 2>/dev/null
 
-.PHONY: install-test-vectors
-install-test-vectors: uninstall-test-vectors
-	mkdir -p $(TEST_VEC_DIRS)
-	wget https://csrc.nist.gov/groups/STM/cavp/documents/mac/kwtestvectors.zip
-	unzip kwtestvectors.zip -d $(TEST_DATA_DIR)
-	rm kwtestvectors.zip
-	wget https://csrc.nist.gov/groups/STM/cavp/documents/mac/gcmtestvectors.zip
-	unzip gcmtestvectors.zip -d $(TEST_DATA_DIR)/gcmtestvectors/
-	rm gcmtestvectors.zip
-
-.PHONY: uninstall-test-vectors
-uninstall-test-vectors:
-	rm -rf $(TEST_VEC_DIRS)
-
 .PHONY: pretest
 pretest:
 	indent $(INDENT_OPTS) $(TEST_SOURCES)
 	indent $(INDENT_OPTS) $(TEST_HEADERS)
-	rm -f test/src/*.c~
-	rm -f test/src/*/*.c~
-	rm -f test/include/*.h~
-	rm -f test/include/*/*.h~
-	mkdir -p $(TEST_OBJECT_DIRS)
 
-kmyth-test: $(TEST_OBJECTS)
+.PHONY: kmyth-test
```

Comment:
Rather than making the target phony would it be better to make the util lib a prerequisite for the test target?

Reply:
This is now $(BIN_DIR)/kmyth-test instead of the phony kmyth-test target.

Reply:
I realized after looking at this again, I missed the part of this comment about adding the library as a prerequisite for the $(BIN_DIR)/kmyth-test target.

Now, if you call *make test* before building, it will build the code, test code, and run the tests.

In doing this, I also realize that if $(LIB_DIR)/libkmyth-logger.so is a prerequisite of $(LIB_DIR)/libkmyth-tpm.so, then by making the TPM utility library a pre-req, the logger library is indirectly a pre-req. Thus, I cleaned up a few places where I listed both libraries as pre-reqs.